src/server/DeviceServer.js

```diff
@@ -87,9 +89,26 @@ class DeviceServer {
     this._eventPublisher = eventPublisher;
     this._areSystemFirmwareAutoupdatesEnabled =
       areSystemFirmwareAutoupdatesEnabled;
+
+    // Master listens to events from workers and broadcast them
+    // to all other workers except the owner.
+    if (useCluster && cluster.isMaster) {
```

Comment:
Ahh.. ok, let's move all of it to the event publisher. I think you'll be able to make things much cleaner in there.